youtube_dl/extractor/twitch.py

```diff
@@ -456,58 +456,49 @@ class TwitchAllVideosIE(TwitchVideosBaseIE):
 
 class TwitchUploadsIE(TwitchVideosBaseIE):
     IE_NAME = 'twitch:videos:uploads'
-    _VALID_URL = r'%s/uploads' % TwitchVideosBaseIE._VALID_URL_VIDEOS_BASE
+    _VALID_URL = r'%s\/?\?.*filter=uploads.*' % TwitchVideosBaseIE._VALID_URL_VIDEOS_BASE
```

Comment:
Do not escape slash. Fix code duplication. Remove useless .* from the end. Do not use greedy mask.

Reply:
* What code duplication ?
* The end `.*` is useful, because the url is actually looking like `https://www.twitch.tv/spamfish/videos?filter=archives&sort=time`. Should I change the test and include the "full" url ?

Reply:
filter.
Again: .* at the end does not make any sense since https://www.twitch.tv/spamfish/videos?
filter=archives&sort=time will match it without it.

Reply:
* Do you mean I shouldn't repeat `%s/?\?.*filter=something.*'` ? (In this case, I'd still have to define another class variable and override the match method to generate a pattern on-the-fly)
* I actually tried this, but it didn't work. `re.match` match for the whole pattern. Anything appended to `filter=archives` would not match if the pattern was `.*filter=archives`. This is a naïve pattern that I have to change (the `.*` prefix is ugly, `\b` would be better for example).

Reply:
>`re.match` match for the whole pattern. Anything appended to `filter=archives` would not match if the pattern was `.*filter=archives`.

Wrong.

Reply:
Indeed, fixed it. A commit is on its way.